src/constants/types/APITypes.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * API 호출 결과 (성공, 실패) 를 나타내는 type
+ * @param T 요청 성공 시 인계 받은 데이터 타입
+ */
+export type APIResponse<T> = Promise<
+  { isSuccess: true; result: T } | { isSuccess: false; result: APIError }
+>;
+
+/**
+ * API 호출 과정에서 발생한 에러와 관련된 interface
+ * @param code 백엔드 측에서 전송한 HTTP Status Code
+ * @param msg 백엔드 측에서 전송한 에러 관련 메세지
+ * @param data 백엔드 측에서 전송한 에러 관련 데이터
+ */
+export interface APIError {
+  code: number;
+  msg: string;
+  data?: any;
+}
```

Comment:
요부분은 함수의 매개변수를 나타내는 `@param` 보다는 요렇게 주석을 다는게 좋을 것 같아요!

```suggestion
export interface APIError {
  /** 백엔드 측에서 전송한 HTTP Status Code */
  code: number;
  /** 백엔드 측에서 전송한 에러 관련 메세지*/
  msg: string;
  /** 백엔드 측에서 전송한 에러 관련 데이터 */
  data?: any;
}
```

Reply:
오... 그렇군요, 확실히 `@param` 의 경우에는 함수의 매개변수를 의미하는 터라 interface는 어떻게 작성해야 할지 잘 몰랐습니다.
앞으로 다른 프로젝트에서도 인터페이스 관련 주석을 달 때는 다음과 같이 작성을 해야겠네요. 알려주셔서 감사합니다.